### PR TITLE
[32059] Update classes and interfaces and mark them as API part 3

### PR DESCRIPTION
**Pull request by Usik2203**
### Description (*)

This PR fixes https://github.com/magento/magento2/issues/32059
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/32059

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Usik2203** [comment]
@magento run all tests

**Comment by BarnyShergold** [comment]
@magento run Unit Tests,WebAPI Tests

**Comment by Usik2203** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run Database Compare, Functional Tests B2B, Functional Tests EE, Semantic Version Checker, Unit Tests

**Comment by eduard13** [comment]
@magento run Database Compare, Functional Tests B2B, Semantic Version Checker, Unit Tests

**Comment by Usik2203** [comment]
@magento run Semantic Version Checker, Unit Tests

**Review by eduard13 — CHANGES_REQUESTED** [review]
hi @Usik2203, could you please check what's reason of Unit Test failing?

**Comment by Usik2203** [comment]
@magento run all tests

File: app/code/Magento/Catalog/Model/Config.php [hunk 1]
```diff
@@ -395,7 +411,7 @@ public function getProductTypeName($id)
      * @param string $value
      * @return null|mixed
      */
-    public function getSourceOptionId($source, $value)
+    public function getSourceOptionId($source, string $value)
```
Comment (eduard13) [inline app/code/Magento/Catalog/Model/Config.php]: @Usik2203 lets revert this change.
```suggestion
    public function getSourceOptionId($source, $value)
```

**Comment by Usik2203** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run Magento Health Index

**Comment by eduard13** [comment]
@magento run Functional Tests B2B

**Review by eduard13 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8879](https://jira.corp.magento.com/browse/ENGCOM-8879) has been created to process this Pull Request

**Comment by Usik2203** [comment]
Closed in favour of https://github.com/magento/magento2/pull/32738

**Pull request closed without merging**